### PR TITLE
Remove mocha-json-output-reporter

### PR DESCRIPTION
**Pull request by anthony-murphy**
The mocha-json-output-reporter is failing which I was able to determine via trial and error by adjusting the reporters, it also appears to be essentially unsupported. It seems we only use it for inline logs in our ci build output, which shouldn't be necessary as we have the test results separately in the test tab. I think the best solution is to just remove this.

### CODE REVIEW

**Comment by CraigMacomber** [comment]
I don't know if this is part of what makes tests show up in our test results section in CI: would be good to confirm that still works before merging this. I chant check this now as CI is failing, looks like due to trailing commas or something.

**Comment by anthony-murphy** [comment]
> I don't know if this is part of what makes tests show up in our test results section in CI: would be good to confirm that still works before merging this. I can't check this now as CI is failing, looks like due to trailing commas or something.

fixed a couple things. also looks like biome is doing bad stuff as well  :(

**Comment by tylerbutler** [comment]
Build failure should be addressed by running `pnpm i --no-frozen-lockfile` in the common/build/eslint-plugin-fluid folder.

File: packages/test/mocha-test-setup/mocharc-common.cjs [hunk 1]
```diff
@@ -76,7 +76,7 @@ function getFluidTestMochaConfig(packageDir, additionalRequiredModules, testRepo
 		`configFile=${path.join(
 			__dirname,
 			"test-config.json",
-		)},cmrOutput=xunit+output+${outputFilePrefix}:mocha-json-output-reporter+output+${outputFilePrefix}:xunit+suiteName+${suiteName}`,
+		)},cmrOutput=xunit+suiteName+${suiteName}`,
```
Comment (Abe27342) [inline packages/test/mocha-test-setup/mocharc-common.cjs:79]: I don't think `cmrOutput` ends up being used. I believe it's an artifact of using [cypress-multi-reporters](https://www.npmjs.com/package/cypress-multi-reporters#cmroutput-option) at some point. I only find reference to `cypress` in property DDS, and it looks incomplete (it's in a mocha.opts file but there's no dependency).

That said, the update here looks wrong. I think we should either remove it fully or remove only the mocha-json-output-reporter options to have

```ts
`cmrOutput=xunit+output+${outputFilePrefix}:xunit+suiteName+${suiteName}`
```

**Review by tylerbutler — APPROVED** [review]

File: packages/test/mocha-test-setup/mocharc-common.cjs (hunk 1 above)
Comment (Abe27342) [reply packages/test/mocha-test-setup/mocharc-common.cjs:79]: `xunit`'s output and suite name look reasonably set in `test-config.json`. I'd vote just remove this and confirm reports still look good in CI

Comment (Abe27342) [reply packages/test/mocha-test-setup/mocharc-common.cjs:79]: oh nevermind, `mocha-multi-reporters` uses the same option name (despite the acronym not being the same), see https://www.npmjs.com/package/mocha-multi-reporters#cmroutput-option. So probably needs an update

Comment (Abe27342) [reply packages/test/mocha-test-setup/mocharc-common.cjs:79]: ohhh yeah I remember how this works now. doc above explains it. The suite name gets substituted in as the value of `{id}` in `test-config.json`. So we do need both, and snippet above specifying the two bits for `xunit` is still required.

File: tools/benchmark/pnpm-lock.yaml [hunk 2]
```diff
@@ -356,7 +353,7 @@ packages:
   /@rushstack/node-core-library@5.3.0(@types/node@14.18.38):
     resolution: {integrity: sha512-t23gjdZV6aWkbwXSE3TkKr1UXJFbXICvAOJ0MRQEB/ZYGhfSJqqrQFaGd20I1a/nIIHJEkNO0xzycHixjcbCPw==}
     peerDependencies:
-      '@types/node': ^18.19.0
+      '@types/node': '*'
```
Comment (alexvy86) [inline tools/benchmark/pnpm-lock.yaml:356]: This bit feels weird and unrelated to the removal, probably revert? Same on line 385.

Comment (anthony-murphy) [inline tools/benchmark/pnpm-lock.yaml:356]:
```suggestion
      '@types/node': '^18.19.0'
```

File: tools/benchmark/pnpm-lock.yaml [hunk 3]
```diff
@@ -356,7 +353,7 @@ packages:
   /@rushstack/node-core-library@5.3.0(@types/node@14.18.38):
     resolution: {integrity: sha512-t23gjdZV6aWkbwXSE3TkKr1UXJFbXICvAOJ0MRQEB/ZYGhfSJqqrQFaGd20I1a/nIIHJEkNO0xzycHixjcbCPw==}
     peerDependencies:
-      '@types/node': ^18.19.0
+      '@types/node': '^18.19.0'
```
Comment (anthony-murphy) [inline tools/benchmark/pnpm-lock.yaml:356]:
```suggestion
      '@types/node': ^18.19.0
```

File: tools/benchmark/pnpm-lock.yaml [hunk 4]
```diff
@@ -382,7 +379,7 @@ packages:
   /@rushstack/terminal@0.12.2(@types/node@14.18.38):
     resolution: {integrity: sha512-yaHKyD/l6Zg34pC5zzc/KdiRBHy8zAH7ZbL3umpDLnvTrZ0SP8MVYZu9xA2lRsGkKfGbv/6gQhyNq4/tRzXH4A==}
     peerDependencies:
-      '@types/node': ^18.19.0
+      '@types/node': '^18.19.0'
```
Comment (anthony-murphy) [inline tools/benchmark/pnpm-lock.yaml:382]:
```suggestion
      '@types/node': ^18.19.0
```

**Pull request merged**